### ISSUE TITLE
Re-enable per-run dataset statistics logging (log_dataset_statistics) for multi-dataset training

### ISSUE THREAD

**Issue by yeelauren** [data-mermaid/mermaid-segmentation#139]
## Summary

`scripts/train.py` used to log per-run dataset QA statistics to MLflow via:

```python
logger.log_dataset_statistics({"train": train_ds, "val": val_ds, "test": test_ds}, registry)
```

(the per-run dataset statistics feature from #112).

The multi-dataset refactor in the config-restructure PR (3/3) builds datasets as **lists** keyed by `(dataset, split)` (`train_datasets` / `val_datasets`), not single `train_ds` / `val_ds` / `test_ds`. `Logger.log_dataset_statistics` expects single per-split datasets, so the call was **commented out** rather than adapted — silently dropping dataset-statistics logging that the single-dataset path had.

## Fix

Adapt `Logger.log_dataset_statistics` (and its call site in `scripts/train.py`) to the multi-dataset structure — e.g. accept the per-split dataset lists or the combined registry datasets — and re-enable the MLflow logging.

## Context

Deferred from the config-restructure PR (3/3); tracked here so the observability gap isn't silent. The call site carries a `# TODO(#<this issue>)` pointing here. Part of finishing the multi-dataset training wiring.

**Comment by josauder** [comment]
Good catch. We would have to do per-dataset statistics, or at least one for sparse and one for dense if we wanted to be correct. For example, a single 512x512 densely annotated image can have 262144 annotated pixels, and so computing one global accuracy will lead to an issue where the global accuracy is essentially just the accuracy on the dense datasets, and all the sparse datasets (the majority of *images*) will not be captured. 

IMO, for the CBM models the best would be to have metrics for at least:
```
train/
    sparse/
          mermaid_classification_accuracy
          multi_hot_accuracy
          taxonomy__kingdom__accuracy
          taxonomy__phylum__accuracy
          ...
          taxonomy__genus__accuracy
          taxonomy__kingdom__accuracy__exclude__none
          taxonomy__phylum__accuracy__exclude__none
          ...
          taxonomy__genus__accuracy__exclude__none
    dense/
          <same>
val/
    sparse/
          <same>          
    dense/
          <same>
```

Previously, I had an issue where MLFlow would crash in my browser, I presume this was because there were too many metrics (we had one accuracy for each of the classes and concepts I believe, so like 800 metrics). Like this, we would have ~60 metrics logging to MLflow

We should likely separate between these validation metrics (cheap to compute, don't need the entire dataset to compute them), and full-on test evaluation, where we would have more advanced metrics evaluated on the whole validation split

**Comment by yeelauren** [comment]
Ah yeah that's not ideal, initially the logged metrics were very broad so probably needs a re-think.
 We can absolutely par down the metrics in that case or at least have some easy levers to turn on/off for specific runs. Is that why the local write to a table was implemented?
